sdks/python/apache_beam/runners/direct/transform_evaluator.py

```diff
@@ -874,6 +875,7 @@ def process_element(self, element):
 
   def finish_bundle(self):
     self.runner.finish()
+    self.runner.teardown()
```

Comment:
Does the runner re-instantiate the DoFn for each bundle? That is the contract of teardown. The DoFn instance can never be touched again. (I'm mostly trying to learn more about the Python SDK and runners)

Reply:
Yes, it does re-instantiate for each bundle. I remember we had a jira somewhere to re-use dofn instances for more than one bundle, we never implemented that.

Reply:
I need to check this, but in Python, generally, we do not reuse Dofns, so they're recreated for each bundle. Let me confirm..